api/src/controllers/theme.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * This module contains handlers for the "theme" route.
+ * @module theme
+ */
+const Theme = require('../models/editor/theme.model');
+
+/**
+ * 
+ * @param {*} req 
+ * @param {*} res 
+ * @returns 
+ */
+async function findOrCreateTheme(req, res) {
+  const themeData = req.body;
+  try {
+    const foundTheme = await Theme.findOne(themeData);
+    if (!foundTheme) {
+      const createdTheme = await Theme.create(themeData);
```

Comment:
Just want to clarify, why the user is able to select a theme that is not currently existing in db? 
Do we allow them to choose their custom color combination?

Reply:
No, just because we have 3 layouts and 3 colour combinations to choose from, and we don't want to put them into the database manually before users selecting their themes. Also if we decided to change or add more options, there will be more combination to put in the database.

Reply:
Got it !